iiif-server/src/index.js

```diff
@@ -47,22 +51,92 @@ function parsePath(path) {
       filename: segments[0]
     };
   } else {
+    const filename = segments[0].split(".");
     return {
       poster: segments[5] == "posters",
       id: segments[4],
       region: segments[3],
       size: segments[2],
       rotation: segments[1],
-      filename: segments[0]
+      filename: segments[0],
+      quality: filename[0],
+      format: filename[1]
     };
   }
 }
 
+function getAuthSignature(request) {
+  if (!request.querystring) {
+    return null;
+  }
+
+  const parsedQuery = querystring.parse(request.querystring);
+  return parsedQuery['Auth-Signature'] || null;
+}
+
+function validateJwtClaims(jwtClaims, params) {
+  const currentTime = Math.floor(Date.now() / 1000);
+  const errors = [];
+
+  if (jwtClaims.exp <= currentTime) {
+    errors.push("Token expired");
+  }
+
+  if (jwtClaims.sub !== params.id) {
+    errors.push("ID mismatch");
+  }
+
+  const fields = ['region', 'size', 'rotation', 'quality', 'format'];
+  for (const field of fields) {
+    if (jwtClaims[field] && !jwtClaims[field].includes(params[field])) {
+      errors.push(`${field} mismatch`);
+    }
+  }
+
+  if (jwtClaims['max-width'] && params.size.width > jwtClaims['max-width']) {
+    errors.push("Width too large");
+  }
+
+  if (jwtClaims['max-height'] && params.size.height > jwtClaims['max-height']) {
+    errors.push("Height too large");
+  }
```

Comment:
`max-width` and `max-height` aren't being enforced, because `params.size` isn't being parsed into width and height. It looks like `2048,1536` or `!300,300` or even `,768` or `pct:50`. This is where all the stuff about the “full frame image” came from in the documentation of those fields – there was a [whole discussion](https://github.com/samvera/serverless-iiif/discussions/152) in the `serverless-iiif` repo about how to handle max dimensions in light of the fact that a savvy and patient user could simply request individual tiles that didn't violate the maximum and stitch them together.

The `iiif-processor` gem offers a way to calculate the effective width and height of the full image based on the parameters given even when a region is requested, but all things considered, I'm not convinced it's worth going to great lengths to support it right now. Maybe the best thing to do is remove `max-width` and `max-height` from the code and the docs and come back to it later if we feel the need. I'm willing to be persuaded otherwise.

Reply:
If you decide to remove the `max-*` support, consider the PR approved once that's done. Otherwise, I'm happy to pair on the rest of implementation if we have time.